### ISSUE TITLE
The current directory under a symlink

### ISSUE THREAD

**Issue by herberteuler** [NixOS/nix#10397]
**Describe the bug**

I ran into this issue while trying to set up a symlink as an overlay. The document says a directory named `overlays` under `~/.config/nixpkgs` is one way to set up overlays, but this behavior precludes symlinks from the solution space as the check on `homeDir + "/.config/nixpkgs/overlays"` being a directory in `impure.nix` always fails.

```
$ mkdir a
$ ln -s a b
$ ls -l
total 4
drwxr-xr-x 2 xgp users 4096 Apr  4 06:48 a
lrwxrwxrwx 1 xgp users    1 Apr  4 06:48 b -> a
$ [ -e b/. ]
$ echo $?
0
$ nix repl
Welcome to Nix 2.18.1. Type :? for help.

nix-repl> builtins.pathExists "/home/xgp/a/b/." 
false
```

**Steps To Reproduce**

Please see the above journals.

**Expected behavior**

`pathExists` conforms with other runtimes.

**`nix-env --version` output**

`nix-env (Nix) 2.18.1`

**Additional context**

**Priorities**

Low, because I switched to use `overlays.nix` as the solution.

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by herberteuler** [comment]
I can work on this one if we agree this needs a fix.

**Comment by thufschmitt** [comment]
That definitely looks like it needs a fix, yes :)

It seems related to the recent symlink overly-eager check thing. @Ericson2314 @edolstra, do you confirm?

**Comment by herberteuler** [comment]
Fixed in at least Nix 2.22.1.
